float64 x_mean
float64 y_mean
float64 z_mean
float64 z_max
float64 z_min
